geomstats/geometry/general_linear.py

```diff
@@ -198,50 +198,9 @@ class SquareMatrices(MatrixLieAlgebra):
         Integer representing the shape of the matrices: n x n.
     """
 
-    def __init__(self, n, equip=True):
+    def __init__(self, n, equip=True, equip_maximally=True):
         self.n = n
-        super().__init__(dim=n**2, representation_dim=n, equip=equip)
-        self._mat_space = Matrices(n, n, equip=False)
+        super().__init__(n=n, m=n, equip=equip, equip_maximally=equip_maximally)
 
-    @staticmethod
-    def default_metric():
-        """Metric to equip the space with if equip is True."""
-        return MatricesMetric
-
-    def _create_basis(self):
-        """Create the canonical basis of the space of matrices."""
-        return self._mat_space.basis
-
-    def basis_representation(self, matrix_representation):
-        """Compute the coefficient in the usual matrix basis.
-
-        This simply flattens the input.
-
-        Parameters
-        ----------
-        matrix_representation : array-like, shape=[..., n, n]
-            Matrix.
-
-        Returns
-        -------
-        basis_representation : array-like, shape=[..., dim]
-            Representation in the basis.
-        """
-        return self._mat_space.flatten(matrix_representation)
-
-    def matrix_representation(self, basis_representation):
-        """Compute the matrix representation for the given basis coefficients.
-
-        This simply reshapes the input into a square matrix.
-
-        Parameters
-        ----------
-        basis_representation : array-like, shape=[..., dim]
-            Coefficients in the basis.
-
-        Returns
-        -------
-        matrix_representation : array-like, shape=[..., n, n]
-            Matrix.
-        """
-        return self._mat_space.reshape(basis_representation)
+    def default_maximal_structures(self):
```

Comment:
--> `default_structures`

we don't need the "maximal" here.

Can you add the docstring and explain there that the default structures are the maximal structures we can equip this space with?